packages/server/graphql/public/mutations/joinTeam.ts

```diff
@@ -0,0 +1,71 @@
+import TeamMemberId from '../../../../client/shared/gqlIds/TeamMemberId'
+import {SubscriptionChannel} from '../../../../client/types/constEnums'
+import acceptTeamInvitationSafe from '../../../safeMutations/acceptTeamInvitation'
+import {analytics} from '../../../utils/analytics/analytics'
+import {getUserId, isTeamMember} from '../../../utils/authorization'
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import {MutationResolvers} from '../resolverTypes'
+import getIsUserIdApprovedByOrg from './helpers/getIsUserIdApprovedByOrg'
+
+const joinTeam: MutationResolvers['joinTeam'] = async (
+  _source,
+  {teamId},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+  const viewerId = getUserId(authToken)
+
+  if (!viewerId) {
+    return standardError(new Error('You must be logged in to join a team'))
+  }
+
+  if (isTeamMember(authToken, teamId)) {
+    return standardError(new Error('You are already a member of this team'))
+  }
+
+  const team = await dataLoader.get('teams').loadNonNull(teamId)
+  const {orgId} = team
+  const viewer = await dataLoader.get('users').loadNonNull(viewerId)
+
+  const hasPublicTeamsFlag = await dataLoader
+    .get('featureFlagByOwnerId')
+    .load({ownerId: orgId, featureName: 'publicTeams'})
+
+  if (!hasPublicTeamsFlag) {
+    return standardError(new Error('This team is not public'))
+  }
```

Comment:
+1 not necessarily part of this change, but I think we might have an issue somewhere to allow org admins to join a team. Allowing them to bypass this check could solve this maybe.